zerver/views/messages.py

```diff
@@ -222,7 +222,7 @@ def by_stream(self, query, operand, maybe_negate):
             # that specific stream.  So it would be a bug to hit this
             # codepath after relying on this term there.  But all streams in
             # a Zephyr realm are private, so that doesn't happen.
-            assert(not stream.is_public())
+            assert(not stream.has_public_history)
```

Comment:
I think this might be better written as `assert stream.is_in_zephyr_realm` or something?